agent.go

```diff
@@ -52,6 +52,11 @@ type bindingRequest struct {
 	destination   net.Addr
 }
 
+type nominatedPair struct {
```

Comment:
Can we use the validPair structure here instead?

**Controlling Side**
Maintain a list of pairs that we saw a full Request/Response for. When we get a request (and are about to send a response) for a pair in this list it becomes our `selectedPair`

**Controlled Side**
validPairs is just the list of everything we did a Request+Response for, and we keep hitting them until we get a `USE-CANDIDATE`

Reply:
yes, we are actually using it like that, but this addional nominatedPair allow us to:

a) check the transaction without worring about the pendingTransaction structure (because we shouldn't risk to loose this tx in case of overlow)
b) In ase we make a nomination in the controlling side we need to differentiate between a normal success reponse and a response to a use candidate (in the later we select the pair)
b) In case we get a nomination in the controlled side for a pair that is not validate yet, we need to start another cycle of request/response and then answer the nomination.

I'm working in another commit, maybe it will be a bit clearer after a little refactor